src/EventStore.Core.Tests/Services/Transport/Tcp/when_invalid_data_is_sent_over_tcp.cs

```diff
@@ -43,23 +42,20 @@ public class when_invalid_data_is_sent_over_tcp<TLogFormat, TStreamId> : specifi
 					new TcpClientConnector(),
 					TimeSpan.FromSeconds(5),
 					(conn) => connectedEvent.Set(),
-					(conn, error) => {
-						Assert.Fail($"Connection failed: {error}");
-					},
+					(conn, error) => failedError = error,
 					false);
 			}
 
-			connection.ConnectionClosed += (conn, error) => {
-				closedEvent.Set();
-			};
+			connection.ConnectionClosed += (conn, error) => closedEvent.Set();
 
-			connectedEvent.WaitOne();
+			Assert.True(connectedEvent.Wait(10000));
 			var data = new List<ArraySegment<byte>> {
 				new ArraySegment<byte>(new byte[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10})
 			};
 			connection.EnqueueSend(data);
-			closedEvent.WaitOne();
+			Assert.True(closedEvent.Wait(10000));
 			connection.Close("intentional close");
+			Assert.Null(failedError);
```

Comment:
same here with respect to threads and taskcompletionsource

Reply:
Changed it.